src/tunnel.cc

```diff
@@ -1062,6 +1084,8 @@ TunnelStateData::connectDone(const Comm::ConnectionPointer &conn, const char *or
         ResetMarkingsToServer(request.getRaw(), *conn);
     // else Comm::ConnOpener already applied proper/current markings
 
+    // TODO: add pconn race state tracking
```

Comment:
Another large out-of-scope change. FwdState has quite a bit of complex code dedicated to properly handle persistent connection races!